### ISSUE TITLE
[Enhancement] Decomposition Core doesn't improve processing of certain items

### ISSUE THREAD

**Issue by grompe** [OreCruncher/ThermalRecycling#41]
I noticed what is a bit odd, that items which don't split into components always get scrapped to same items, with same chances, regardless of Decomposition Core present in the upgrade slot:
50% lost, 1.5% current level scrap (if any), 0.5% higher level scrap, rest — debris.

Is this intended?

**Comment by OreCruncher** [comment]
Yes.  An item that doesn't have component parts gets scrapped as if there wasn't a core inside.  Currently it is working as intended.  I do want to change this, but right now it would require changes in sensitive areas.

**Comment by grompe** [comment]
Small addition: presence of the core actually slows down the process in such case. Which brings a thought: what if certain results are best achieved with certain level of Core?

**Comment by OreCruncher** [comment]
Yes, it does.  A recycler running without a core (straight out scrap mode) is twice as fast as one with a decomp core.  Also, a recycler with an extraction core is about twice as slow as a decomp core.

As for differing results that is why there are 5 levels of core: Basic, Hardened, Reinforced, Resonant, and Ethereal.  Each level of core increases the keep chance and reduces the dust chance of a component.  Ethereal is 100% retain.
